web/components/elements/ConnectionSearch.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useMemo, useState } from "preact/hooks";
+import { BridgeAPIError } from "../../BridgeAPI";
+import { DropdownSearch, DropItem } from "./DropdownSearch";
+import { ErrorPane } from "./ErrorPane";
+import { InputField } from "./InputField";
+
+interface Instance { 
+    name: string;
+}
+
+type Project = DropItem;
+
+interface IProps {
+    serviceName: string;
+    getInstances(): Promise<Instance[]>;
+    getProjects(currentInstance: string, searchTerm?: string): Promise<Project[]>;
+    onPicked: (instanceValue: string, projectValue: string) => void;
+    onClear: () => void;
+}
+
+/**
+ * This component is designed to generically fetch a bunch of instances for a given connection type
+ * and then a list of projects associated with that instance. The user should be able to pick from
+ * that list.
+ * @param props 
+ * @returns 
+ */
+export function ConnectionSearch({
+    serviceName,
+    onPicked,
+    onClear,
+    getInstances,
+    getProjects,
+}: IProps) {
+    const [currentInstance, setCurrentInstance] = useState<string>("");
+    const [instances, setInstances] = useState<Instance[]|null>(null);
+    const [searchError, setSearchError] = useState<string|null>(null);
+    const [exampleProjectName, setExampleProjectName] = useState<string>("Loading...");
+
+    useEffect(() => {
+        getInstances().then(res => {
+            setInstances(res);
+            setCurrentInstance(res[0]?.name ?? null);
```

Comment:
This state got defined as a string with an empty string being the default value.
Good morning, TypeScript linter. :wave: 

```suggestion
            setCurrentInstance(res[0]?.name ?? '');
```